### PR TITLE
Feature/mip container

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### This PR adds | fixes:

- Adds a full mip container to use when not interacting with SLURM
- Add github actions to build and push docker mip image
- Rename MIP base image to mip_base to avoid namespace clash in dockerhub and bump its version
- Use new MIP base image in the rest of the MIP dockerfiles
- Use new mip container in MIP specific recipes
- Remove sourcing of conda env form these MIP specific recipes

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [ ] Installation, unit and integration test suite pass

### Review:
- [ ] Code review
- [ ] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [x] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by henrikstranneheim** [comment]
```
cat /home/henrik.stranneheim/develop/livingox/analysis/scripts/livingox/qccollect_ar/dry_run_qccollect_ar_livingox.9.sh
#! /bin/bash --login
#SBATCH --account=development
#SBATCH --ntasks=1
#SBATCH --time=1:00:00
#SBATCH --mem=5G
#SBATCH --qos=low
#SBATCH --job-name=qccollect_ar_livingox
#SBATCH --error=/home/henrik.stranneheim/develop/livingox/analysis/livingox/qccollect_ar/info/dry_run_qccollect_ar_livingox.9.stderr.txt
#SBATCH --output=/home/henrik.stranneheim/develop/livingox/analysis/livingox/qccollect_ar/info/dry_run_qccollect_ar_livingox.9.stdout.txt
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=henrik.stranneheim@scilifelab.se

set -e
set -o pipefail
readonly PROGNAME=$(basename "$0")

echo "Running on: $(hostname)"

## Create temporary directory
readonly TEMP_DIRECTORY=/home/henrik.stranneheim/develop/livingox/analysis/"$SLURM_JOB_ID"
mkdir -p "$TEMP_DIRECTORY"

finish() {

	local directory="$1"
	## Perform exit housekeeping
	rm -R -f "$directory"

}

## Enable trap for signal(s) EXIT TERM INT
trap '$(finish "$TEMP_DIRECTORY")' EXIT TERM INT

XDG_RUNTIME_DIR="$TEMP_DIRECTORY"

## Enable trap for signal(s) DEBUG
trap 'previous_command="$BASH_COMMAND"' DEBUG

error() {

	local program="$1"

	local return_code="$2"

	## Display error message and exit
	echo "${program}: ${return_code}: Unknown Error - ExitCode=$return_code" 1>&2
	exit 1
}

## Enable trap for signal(s) ERR
trap '$(error"$previous_command" "$?")' ERR

singularity exec --bind /home/proj/development/rare-disease/references_9.0:/home/proj/development/rare-disease/references_9.0,/home/henrik.stranneheim/develop/livingox/analysis:/home/henrik.stranneheim/develop/livingox/analysis docker://docker.io/clinicalgenomics/mip:9.0.5 mip qccollect --eval_metric_file /home/proj/development/rare-disease/references_9.0/qc_eval_metric_-v1.3-.yaml --log_file /home/henrik.stranneheim/develop/livingox/analysis/livingox/qccollect_ar/livingox_qc_metrics_qccollect.log --outfile /home/henrik.stranneheim/develop/livingox/analysis/livingox/qccollect_ar/livingox_qc_metrics.yaml --regexp_file /home/proj/development/rare-disease/references_9.0/qc_regexp_-v1.27-.yaml --sample_info_file /home/henrik.stranneheim/develop/livingox/analysis/livingox_qc_sample_info.yaml
```

```
[0|0|790] [henrik.stranneheim@hasta:~] [D_mip_hs] 4s $ sbatch /home/henrik.stranneheim/develop/livingox/analysis/scripts/livingox/qccollect_ar/dry_run_qccollect_ar_livingox.9.sh
Submitted batch job 733412
```
**PASS**
```
[0|0|785] [henrik.stranneheim@hasta:~] [D_mip_hs] 3s $ ejobinfo 733412 | cat
[INFO] 2020/11/13 15:28:35 MIP_QCCOLLECT - Loading: /home/henrik.stranneheim/develop/livingox/analysis/livingox_qc_sample_info.yaml
[INFO] 2020/11/13 15:28:35 MIP_QCCOLLECT - Loaded: /home/henrik.stranneheim/develop/livingox/analysis/livingox_qc_sample_info.yaml
[INFO] 2020/11/13 15:28:35 MIP_QCCOLLECT - Loading: /home/proj/development/rare-disease/references_9.0/qc_regexp_-v1.27-.yaml
[INFO] 2020/11/13 15:28:35 MIP_QCCOLLECT - Loaded: /home/proj/development/rare-disease/references_9.0/qc_regexp_-v1.27-.yaml
[INFO] 2020/11/13 15:28:58 MIP_QCCOLLECT - Wrote: /home/henrik.stranneheim/develop/livingox/analysis/livingox/qccollect_ar/livingox_qc_metrics.yaml
```

**Comment by henrikstranneheim** [comment]
However, vercollect is trying to call a container from a container in this solution rigth now and that will probably only work with podman...
I could revert this particular recipe and await podman api...

**Comment by jemten** [comment]
Will it work with docker?

**Comment by henrikstranneheim** [comment]
I do not know - have to give it a spin or read about it to know. I am leaning towards reversing this particular recipe and program to deal with this one in a later more specific PR

File: t/data/test_data/reference_info.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
 mip:
-  path: /path/MIP/t/data/modules/miniconda/envs/mip_ci/bin/mip
+  path: /Users/henrik.stranneheim/git/MIP/t/data/modules/miniconda/envs/mip_ci/bin/mip
```
Comment (henrikstranneheim) [inline t/data/test_data/reference_info.yaml:3]:
```suggestion
  path: /path/MIP/t/data/modules/miniconda/envs/mip_ci/bin/mip
```

**Comment by henrikstranneheim** [comment]
@jemten and @ramprasadn Could you give this another review? I have changed mip_vercollect now to work with this new set-up

File: .github/workflows/docker_build_n_publish.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
+
+name: Publish to Docker Hub
```
Comment (jemten) [inline .github/workflows/docker_build_n_publish.yml:3]: Nice, with automated builds!

**Review by jemten — APPROVED** [review]
Looks great!

**Pull request merged**